internal/worker_version_sets.go

```diff
@@ -130,6 +155,35 @@ type GetWorkerBuildIdCompatibilityOptions struct {
 	MaxSets   int
 }
 
+type GetWorkerTaskReachabilityOptions struct {
+	// BuildIDs - The build IDs to query the reachability of. At least one build ID must be provided.
+	BuildIDs []string
+	// TaskQueues - The task queues with Build IDs defined on them that the request is
+	// concerned with.
+	// Optional: defaults to all task queues
+	TaskQueues []string
+	// Reachability - The reachability this request is concerned with.
+	// Optional: defaults to all types of reachability
+	Reachability TaskReachability
+}
+
+type WorkerTaskReachability struct {
+	// Reachable - map of build IDs and their reachability information
+	// May contain an entry with [UnversionedBuildID] for an unversioned worker
+	Reachable map[string]*BuildIDReachability
+}
+
+type BuildIDReachability struct {
+	// TaskQueueReachable map of task queues and their reachability information
+	TaskQueueReachable map[string]*TaskQueueReachability
+}
+
+type TaskQueueReachability struct {
```

Comment:
Some SDKs have a `UnretrievedTaskQueues` here. Not sure why TS doesn't.

Reply:
hm yeah, I based the design on what TS did. I prefer what Python has with an explicate container for unfetched task queues. @Sushisource thoughts?

Reply:
The more explicit version is preferable I think. In TS it works OK because the empty string is assigned a special type, but that's not workable here.

Reply:
Might be worth issuing multiple requests to populate all of the unretrieved task queues instead of having the user do it.

Reply:
If we did that, then the API should've just done all that anyways to begin with. But we didn't, because it could be overloading sometimes. I think we should either respect that, or just not have the limitation.

Reply:
The API has limits in place to prevent fanout for a single request, and we also have rate limits on visibility APIs like this one. Users may issue follow-up requests to fill the missing data and they'll stay within the system limits.

Whether or not we make multiple calls on behalf of users can be configurable but I think experience-wise we'd prefer defaulting to giving all the information and hiding this detail from users. That's of course my personal take on this, I understand you have a different one.

In any case, we can add this later as the feature matures and we get feedback.